x/concentrated-liquidity/lp.go

```diff
@@ -80,6 +80,16 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	// TODO: Utilize the position ID returned from this function in the position key.
 	k.getNextPositionIdAndIncrement(ctx)
 
+	hasFullPosition := k.hasFullPosition(ctx, poolId, owner, lowerTick, upperTick, joinTime, freezeDuration)
+	if !hasFullPosition {
+		err = k.initPositionUptime(ctx, poolId, owner, lowerTick, upperTick, liquidityDelta, joinTime, freezeDuration)
+		if err != nil {
+			return sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, err
+		}
+	} else {
+		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, time.Time{}, types.PositionNotFoundError{PoolId: poolId, LowerTick: lowerTick, UpperTick: upperTick, JoinTime: joinTime, FreezeDuration: freezeDuration}
```

Comment:
Is there a test covering this?

Also, this should not be a `PositionNotFoundError`. We should have a new one for attempting to add to an existing position

Reply:
yup changed to PositionAlreadyExist Error and is covered here: https://github.com/osmosis-labs/osmosis/pull/4508/files#diff-91569dd5b56ddbfc9450f57de47bc13061e1daceaba154cab15b635d98912d19R1716-R1773

Reply:
The linked test seems to cover another function.

We should have a direct test for `createPosition` in `lp_test.go` for this. I think the commented-out test that was removed can be converted to cover this

Reply:
converted commented out test to cover this 
https://github.com/osmosis-labs/osmosis/pull/4508/files#diff-8a43021241fc488af68dd6cdd7c762a85476f3aa8e44248784e9265c9ef0bbdfR799